internal/worker/caasapplicationprovisioner/ops.go

```diff
@@ -401,101 +444,76 @@ func updateState(
 		if err != nil && !errors.Is(err, applicationerrors.ApplicationNotFound) {
 			return nil, errors.Trace(err)
 		}
-		appStatus = params.EntityStatus{
-			Status: svc.Status.Status,
-			Info:   svc.Status.Message,
-			Data:   svc.Status.Data,
+		now := clk.Now()
+		err = statusService.SetApplicationStatus(ctx, appName, status.StatusInfo{
+			Status:  svc.Status.Status,
+			Message: svc.Status.Message,
+			Data:    svc.Status.Data,
+			Since:   &now,
+		})
+		if err != nil && !errors.Is(err, statuserrors.ApplicationNotFound) {
```

Comment:
question: why is it okay to continue here if the application is not found? Especially calling `SetApplicationStatus` if `UpdateCloudService` failed?